ShopApp/ShopApp/ViewController.swift

```diff
@@ -7,14 +7,64 @@
 
 import UIKit
 import FirebaseDatabase
+import FirebaseAuth
+import FirebaseCore
+import GoogleSignIn
 
 class ViewController: UIViewController {
 
+    @IBOutlet private var signInButton: GIDSignInButton!
+
+    @IBAction private func googleSignIn() {
+        guard let clientID = FirebaseApp.app()?.options.clientID else { return }
+
+        // Create Google Sign In configuration object.
+        let config = GIDConfiguration(clientID: clientID)
+
+        // Start the sign in flow!
+        GIDSignIn.sharedInstance.signIn(with: config, presenting: self) { [unowned self] user, error in
+
+            if error != nil {
+                return
+            }
+
+            guard
+                let authentication = user?.authentication,
+                let idToken = authentication.idToken
+            else {
+                return
+            }
+
+            let credential = GoogleAuthProvider.credential(withIDToken: idToken, accessToken: authentication.accessToken)
+
+            Auth.auth().signIn(with: credential) { _, error in
+
+                if let error = error {
+                    _ = error as NSError
+                    print("Error")
+                } else {
+            // User is signed in
+                    print("signed")
+                    let storyboard = UIStoryboard(name: "HomePage", bundle: nil)
+                    let home = storyboard.instantiateViewController(identifier: "HomeViewController")
+                    let navigationController = self.navigationController
+                    navigationController?.setViewControllers([home], animated: true)                }
+            }
+        }
+    }
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
 
-        let ref = Database.database().reference()
+        if Auth.auth().currentUser != nil {
```

Comment:
Extract this code in a separate method and call it here. It's better practice in order not to overcrowd the viewDidLoad method